### ISSUE TITLE
Undefined variable in inlining pass under Julia 0.3.4

### ISSUE THREAD

**Issue by simonster** [JuliaLang/julia#9630]
Not sure if anyone's going to want to debug this, but when running the tests for [the Lasso package I'm working on](https://github.com/simonster/Lasso.jl/tree/5bd8bb5e91b3fb7ef86a4ec3c10c99a499d6a670) on Julia 0.3.4, I get the following error:

```
ERROR: #s3694 not defined
 in inlining_pass at /Applications/Julia-0.3.4.app/Contents/Resources/julia/lib/julia/sys.dylib (repeats 4 times)
 in typeinf at /Applications/Julia-0.3.4.app/Contents/Resources/julia/lib/julia/sys.dylib
 in typeinf_ext at /Applications/Julia-0.3.4.app/Contents/Resources/julia/lib/julia/sys.dylib
 in anonymous at /Users/simon/.julia/v0.3/Lasso/test/runtests.jl:22
 in facts at /Users/simon/.julia/v0.3/FactCheck/src/FactCheck.jl:286
 in anonymous at no file:21
 in include at /Applications/Julia-0.3.4.app/Contents/Resources/julia/lib/julia/sys.dylib
 in include_from_node1 at loading.jl:128
 in process_options at /Applications/Julia-0.3.4.app/Contents/Resources/julia/lib/julia/sys.dylib
 in _start at /Applications/Julia-0.3.4.app/Contents/Resources/julia/lib/julia/sys.dylib
while loading /Users/simon/.julia/v0.3/Lasso/test/runtests.jl, in expression starting on line 20
```

The tests stochastically pass on master, with failures due to #9232.

**Comment by JeffBezanson** [comment]
Does this also happen with other 0.3 versions?

**Comment by simonster** [comment]
Yes, this happens Julia 0.3.0-0.3.3 as well.

**Comment by ihnorton** [comment]
fixed in #10044
